bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/AbstractArrayValue.java

```diff
@@ -73,6 +75,29 @@ public void append(Object value) {
         add(size, value);
     }
 
+    @Override
+    public boolean equals(Object o, Set<ValuePair> visitedValues) {
+        ValuePair compValuePair = new ValuePair(this, o);
+        for (ValuePair valuePair : visitedValues) {
+            if (valuePair.equals(compValuePair)) {
+                return true;
+            }
+        }
+        visitedValues.add(compValuePair);
+
```

Comment:
Shall we remove unwanted new lines

Reply:
I added it to improve the readability as following lines belongs to a separate logic. Isn't it okay to keep?